includes/api/save-abandonment-data.php

```diff
@@ -49,114 +49,360 @@ function saveCartAbandonmentData(WP_REST_Request $request)
+    $billingFirstName = $razorpayData['customer_details']['billing_address']['name'] ?? '';
+    $billingLastName = " ";
+    $billingZipcode = $razorpayData['customer_details']['billing_address']['zipcode'] ?? '';
+    $shippingZipcode  = $razorpayData['customer_details']['shipping_address']['zipcode'] ?? '';
+
+    $shippingCharges = WC()->cart->shipping_total;
+
+    // Insert record in abandoned cart table for the guest user.
+    $userId = saveGuestUserDetails($billingFirstName, $billingLastName, $razorpayData['customer_details']['email'],$billingZipcode, $shippingZipcode, $shippingCharges);
+
+    wcal_common::wcal_set_cart_session('user_id', $userId);
+    
+    $currentTime = current_time( 'timestamp' ); // phpcs:ignore
+
+    $results =  checkUserIdExist($userId);
+
+    $cart = array();
+
+    $cart['cart'] = WC()->session->cart;
+
+    if (count($results) === 0)
+    {
+        $getCookie = WC()->session->get_session_cookie();
 
-            if (isset($result)) {
-                $wpdb->update(
-                    $cartAbandonmentTable,
-                    $checkoutDetails,
-                    array('session_id' => $sessionId)
-                );
-
-                if ($wpdb->last_error) {
-                    $response['status']  = false;
-                    $response['message'] = $wpdb->last_error;
-                    $statusCode          = 400;
-                } else {
-                    $response['status']  = true;
-                    $response['message'] = 'Data successfully updated for wooCommerce cart abandonment recovery';
-                    $statusCode          = 200;
-                }
-
-            } else {
-                $sessionId                     = md5(uniqid(wp_rand(), true));
-                $checkoutDetails['session_id'] = sanitize_text_field($sessionId);
-
-                // Inserting row into Database.
-                $wpdb->insert(
-                    $cartAbandonmentTable,
-                    $checkoutDetails
-                );
-
-                if ($wpdb->last_error) {
-                    $response['status']  = false;
-                    $response['message'] = $wpdb->last_error;
-                    $statusCode          = 400;
-                } else {
-                    // Storing session_id in WooCommerce session.
-                    WC()->session->set('wcf_session_id', $sessionId);
-                    $response['status']  = true;
-                    $response['message'] = 'Data successfully inserted for wooCommerce cart abandonment recovery';
-                    $statusCode          = 200;
-                }
+        $cartInfo = wp_json_encode($cart);
+        $results =  checkRecordBySession($getCookie[0]);
+
+        if (get_post_meta($wcOrderId, 'abandoned_user_id', true) == '')
+        {
+            add_post_meta( $wcOrderId, 'abandoned_user_id', $userId); }
+        else
+        {
+            update_post_meta( $wcOrderId, 'abandoned_user_id', $userId );
```

Comment:
@poojarazor I think `update_post_meta` will check the key existence, so please confirm whether `add_post_meta` is required or not.